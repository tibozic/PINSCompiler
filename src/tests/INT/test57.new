Exception in thread "main" java.lang.NullPointerException: Cannot read field "label" because "elseLabel" is null
	at compiler.ir.IRCodeGenerator.visit(IRCodeGenerator.java:453)
	at compiler.parser.ast.expr.IfThenElse.accept(IfThenElse.java:73)
	at compiler.ir.IRCodeGenerator.lambda$visit$2(IRCodeGenerator.java:301)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
	at compiler.ir.IRCodeGenerator.visit(IRCodeGenerator.java:299)
	at compiler.parser.ast.expr.Block.accept(Block.java:27)
	at compiler.ir.IRCodeGenerator.visit(IRCodeGenerator.java:617)
	at compiler.parser.ast.expr.Where.accept(Where.java:33)
	at compiler.ir.IRCodeGenerator.visit(IRCodeGenerator.java:644)
	at compiler.parser.ast.def.FunDef.accept(FunDef.java:44)
	at compiler.ir.IRCodeGenerator.lambda$visit$3(IRCodeGenerator.java:629)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
	at compiler.ir.IRCodeGenerator.visit(IRCodeGenerator.java:628)
	at compiler.parser.ast.def.Defs.accept(Defs.java:30)
	at compiler.ir.IRCodeGenerator.visit(IRCodeGenerator.java:616)
	at compiler.parser.ast.expr.Where.accept(Where.java:33)
	at compiler.ir.IRCodeGenerator.visit(IRCodeGenerator.java:644)
	at compiler.parser.ast.def.FunDef.accept(FunDef.java:44)
	at compiler.ir.IRCodeGenerator.lambda$visit$3(IRCodeGenerator.java:629)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
	at compiler.ir.IRCodeGenerator.visit(IRCodeGenerator.java:628)
	at compiler.parser.ast.def.Defs.accept(Defs.java:30)
	at Main.run(Main.java:136)
	at Main.run(Main.java:50)
	at Main.main(Main.java:41)
