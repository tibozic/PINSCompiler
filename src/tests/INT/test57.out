"5 is not larger than 10"
"20 is larger than 10"
"x1 < x2"
true
"x1 >= x2"
true
